allennlp/run.py

```diff
@@ -0,0 +1,128 @@
+from typing import Any, Dict, Union
```

Comment:
This sounds like a good idea. As well as providing scripts, we can use [this pipy functionality](https://packaging.python.org/tutorials/distributing-packages/#console-scripts) to automatically create runnable binaries when pip installing `allennlp`. These could be `allennlp-train` and `allennlp-eval` etc, with the contents of what is in `run.py` as their actual content. @joelgrus have you made pip packages with console scripts before? Maybe there are benefits to doing it another way that I don't know about but this seems straightforward.